sdk/core/azure-core/CMakeLists.txt

```diff
@@ -131,6 +133,10 @@ target_include_directories(
 # make sure that users can consume the project as a library.
 add_library(Azure::azure-core ALIAS azure-core)
 
+if (NOT MSVC)
+  set_source_files_properties(src/base64.cpp PROPERTIES COMPILE_FLAGS -Wno-old-style-cast)
```

Comment:
@jorgen where is this warning coming from, and on what compiler/platform?

Reply:
[root@7b5b85ecf427 /]# gcc --version
gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Reply:
This is coming from the fact that BIO_get_mem_ptr is implemented as a macro, and it does a c-style cast on RedHat 7

Reply:
relevant to discussion https://github.com/Azure/azure-sdk-for-cpp/issues/2346#issuecomment-852638004

I think it'd be better if we use `#pragma GCC diagnostic` to disable this warning just for that one line of code, rather than disable the warning for the whole file.

Reply:
I have changed it for `#pragma GCC diagnostic` now.